### ISSUE TITLE
Development builds can't start with an Athlon XP 2600+ (Windows XP 32 bit)

### ISSUE THREAD

**Issue by Moiry** [miranda-ng/miranda-ng#976]
Hello,

VS 2017 requires the SS2 instruction Set ?

Not available on Athlon XP ?

Thanks

**Comment by georgehazan** [comment]
hi, it shall not use any additional instruction sets
why do you think that it requires SS2?

**Comment by dartraiden** [comment]
>> can't start

Can you provide error message text, please?

**Comment by Moiry** [comment]
Hi,

No error message, no crash dump. It appears in the task manager and closes. It is as if nothing happened.

I have disabled all my resident programs, and it's the same result.

I have tested with the installer and with the update from Miranda (0.95.7.18570 - 22 September 2017 - works correctly). All files (api-ms-win-core-console-l1-1-0.dll...) are present in the Miranda folder.

The stable version works correctly (lastest 0.95.6.18511 - 5 October 2017).


On the same Windows with a Phenom II X2 (just with the detection of the new mainboard/config), Miranda (development version) works normally. 

I thought of a compilation that checks if the SS2 is present, even if it is not used. I'm not a developer. ;-)


PS: sorry for my english, i am french.

**Comment by georgehazan** [comment]
well.... according to the documentation provided by MS, VS2017 can use SSE2 instructions only for 64-bit programs, 32-bit programs shall use only 'blind' instruction set (for Pentium processor). maybe you tried to launch 64-bit version?

**Comment by georgehazan** [comment]
moreover, could you unzip this program to the folder that contains Miranda32.exe and run it?
it's compiled on the same runtime with Miranda NG and simply prints "boo". Will it crash?
[boo.zip](https://github.com/miranda-ng/miranda-ng/files/1363101/boo.zip)

**Comment by Moiry** [comment]
Hi,

I also tested on an Athlon XP 1.3 Ghz motherboard. So I have:

1 Athlon XP 2600+/VIA
1 Athlon XP 1.3 Ghz/nForce
1 Phenom II/AMD 760G

I have re-downloaded:

Setup: https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.exe

7z:        https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.7z

Summary for stable, dev versions and boo.exe:
----------------------------------------------------------

Tests performed with the installation of the Athlon XP 2600+ on the Phenom II and the Athlon XP 1.3 Ghz. Also with a new installation on each PC, without any programs resident...

- Miranda NG stable (0.95.6.18511):

  work on (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+, Phenom II

- Miranda NG Development VS 2017 (0.95.7.18634):

  work on (old and new installations): Phenom II

  do not work (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+

- Boo.exe:

Do not run on any machine

https://bigimage.cz/pt/AF5D/XP1300.html

In englis: "boo.exe is not a valid Win32 application"

- Also tested with "vc_redist.x86-2015-RC3.exe". Same results on Athlon XP 2600+ and 1.3 Ghz: do not run and the same error for "boo.exe".

Thanks

**Comment by Moiry** [comment]
Hi,

I also tested on an Athlon XP 1.3 Ghz motherboard. So I have:

1 Athlon XP 2600+/VIA
1 Athlon XP 1.3 Ghz/nForce
1 Phenom II/AMD 760G

I have re-downloaded:

Setup: https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.exe

7z:        https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.7z

Summary for stable, dev versions and boo.exe:
--------------------------------------------

Tests performed with the Windows installation of the Athlon XP 2600+ on the Phenom II and the Athlon XP 1.3 Ghz. Also with a new Windows installation on each PC, without any programs resident ...

- Miranda NG stable (0.95.6.18511):

  work on (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+, Phenom II

- Miranda NG Development VS 2017 (0.95.7.18634):

  work on (old and new installations): Phenom II

  do not work (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+

- Boo.exe:

Do not run on any machine

https://bigimage.cz/pt/AF5D/XP1300.html

In englis: "boo.exe is not a valid Win32 application"

- Also tested with "vc_redist.x86-2015-RC3.exe". Same results on Athlon XP 2600+ and 1.3 Ghz: do not run and the same error for "boo.exe".

Thanks

**Comment by Moiry** [comment]
Hi,

I also tested on an Athlon XP 1.3 Ghz motherboard. So I have:

1 Athlon XP 2600+/VIA
1 Athlon XP 1.3 Ghz/nForce
1 Phenom II/AMD 760G

I have re-downloaded:

Setup: https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.exe

7z:        https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.7z

Summary for stable, dev versions and boo.exe:
--------------------------------------------

Tests performed with the Windows installation of the Athlon XP 2600+ on the Phenom II and the Athlon XP 1.3 Ghz. Also with a new installation on each PC, without any programs resident ...

- Miranda NG stable (0.95.6.18511):

  work on (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+, Phenom II

- Miranda NG Development VS 2017 (0.95.7.18634):

  work on (old and new installations): Phenom II

  do not work (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+

- Boo.exe:

Do not run on any machine

https://bigimage.cz/pt/AF5D/XP1300.html

In englis: "boo.exe is not a valid Win32 application"

- Also tested with "vc_redist.x86-2015-RC3.exe". Same results on Athlon XP 2600+ and 1.3 Ghz: do not run and the same error for "boo.exe". Boo.exe do not run on Phenom II.

Thanks

**Comment by Moiry** [comment]
Hi,

I also tested on an Athlon XP 1.3 Ghz motherboard. So I have:

1 Athlon XP 2600+/VIA
1 Athlon XP 1.3 Ghz/nForce
1 Phenom II/AMD 760G

I have re-downloaded:

Setup: https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.exe

7z:        https://www.miranda-ng.org/distr/miranda-ng-alpha-latest.7z

Summary for stable, dev versions and boo.exe:
--------------------------------------------

Tests performed with the Windows installation of the Athlon XP 2600+ on the Phenom II and the Athlon XP 1.3 Ghz. Also with a new installation on each PC, without any programs resident ...

- Miranda NG stable (0.95.6.18511):

  work on (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+, Phenom II

- Miranda NG Development VS 2017 (0.95.7.18634):

  work on (old and new installations): Phenom II

  do not work (old and new installations): Athlon XP 1.3 Ghz, Athlon XP 2600+

- Boo.exe:

Do not run on any machine

https://bigimage.cz/pt/AF5D/XP1300.html

In englis: "boo.exe is not a valid Win32 application"

- Also tested with "vc_redist.x86-2015-RC3.exe". Same results on Athlon XP 2600+ and 1.3 Ghz: do not run and the same error for "boo.exe". Boo.exe do not run on Phenom II (same error).

Thanks

**Comment by georgehazan** [comment]
too bad. since **boo** doesn't do anything but printf & return, it means that even 32-bit runtime of VS2017 requires SSE2....

**Comment by Moiry** [comment]
That's what I thought with my first post. ;-) It may not help, but I found it for VS 2015:

" Because the x86 compiler generates code that uses SSE2 instructions by default, you must specify /arch:IA32 to disable generation of SSE and SSE2 instructions for x86 processors. "

https://msdn.microsoft.com/en-us//library/7t5yh4fd.aspx

To test ? Is this valid for VS 2017 ? Normally not according to your answer:

"32-bit programs shall use only 'blind' instruction set (for Pentium processor)"

**Comment by Moiry** [comment]
That's what I thought with my first post. ;-) It may not help, but I found it for VS 2015:

" Because the x86 compiler generates code that uses SSE2 instructions by default, you must specify /arch:IA32 to disable generation of SSE and SSE2 instructions for x86 processors. "

https://msdn.microsoft.com/en-us//library/7t5yh4fd.aspx

To test ? Is this valid for VS 2017 ? Normally not according to your answers:

"32-bit programs shall use only 'blind' instruction set (for Pentium processor)"

"it means that even 32-bit runtime of VS2017 requires SSE2...."

**Comment by georgehazan** [comment]
okay, this is the variant compiled with /arch:IA32
[boo.zip](https://github.com/miranda-ng/miranda-ng/files/1366690/boo.zip)

**Comment by Moiry** [comment]
Miranda Dev version 0.95.7.18664 runs correctly ! Tested on both Athlon XP and Phenom II on all Windows installations.

Boo.exe (version 2) makes the same error on all machines. It does not matter. ;-)

Is it possible to make a donation ? For a while, I have not found anything on the official website and on Google.

Thanks

**Comment by Moiry** [comment]
Miranda Dev version 0.95.7.18664 runs correctly. Tested on both Athlon XP and Phenom II on all Windows installations.

Miranda crashes when checking for available updates.

Boo.exe (version 2) makes the same error on all machines. It does not matter. ;-)

Is it possible to make a donation ? For a while, I have not found anything on the official website and on Google.

**Comment by Moiry** [comment]
Miranda Dev version 0.95.7.18664 runs correctly. Tested on both Athlon XP and Phenom II on all Windows installations.

But, Miranda crashes when checking for available updates.

Boo.exe (version 2) makes the same error on all machines. It does not matter. ;-)

Is it possible to make a donation ? For a while, I have not found anything on the official website and on Google.

Thanks

**Comment by georgehazan** [comment]
nice to hear.
seems that Plugin Updater loads smth that crashes... crash log is also not availlable?
boo.exe requires the same runtime that Miranda uses, you need to launch it from the same folder with Miranda and all those ms-api-blablabla.dll
nope, we currently don't get them, and for what? you did all job for me and found that option :)

**Comment by Moiry** [comment]
Sorry! The plugin updater works correctly. It crashed yesterday, but it is functional today (tested three times). I had to make a error...

For the crash dump, I had deleted the folder, so as not to have too much and mix. Sorry!

The latest Dev version of Miranda (0.95.7.18667 - 9 october 2017 (04:00:38)) works correctly (all CPU/installations). Update has been made since Mirana Dev version 0.95.7.18664 (Plugin Updater).

Boo1.exe and Boo2.exe are in Miranda folder (Dev - 0.95.7.18667) with all files (ms-api-blablabla.dll), same error. Re-tested on XP (clean and old installations), Athlon XP and Phenom II, with and without all x86 C++ redistrib (2010, 2012, 2013, 2015 Update 3 RC ). 

Boo1.exe and Boo2.exe work on Seven Home Basic x64.

"nope, we currently don't get them, and for what? you did all job for me and found that option :)"
OK. ;-)
